archivers/unshield/Portfile

```diff
@@ -1,36 +1,35 @@
-PortSystem			1.0
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
-name				unshield
-version				0.6
-categories			archivers compression sysutils
-license				MIT
-platforms			darwin
-maintainers			nomaintainer
-description			Extract files from InstallShield cabinet files
-
-long_description \
-	Extract files from InstallShield-style cabinet files (usually named \
-	data1.cab, data1.hdr, data2.cab, etc), versions 5 and later
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           cmake 1.1
 
+github.setup        twogood unshield 1.4.3
+github.tarball_from archive
 
-homepage			http://synce.sourceforge.net/synce/unshield.php
-master_sites		sourceforge:synce
+categories          archivers compression sysutils
+license             MIT
+platforms           darwin
+maintainers         nomaintainer
+description         Extract files from InstallShield cabinet files
 
-depends_lib			port:zlib
+long_description \
+    Extract files from InstallShield-style cabinet files (usually named \
+    data1.cab, data1.hdr, data2.cab, etc), versions 5 and later
 
-checksums			md5     31a829192a255160d1f71cda4c865c9c \
-					sha1    3e1197116145405f786709608a5a636a19f4f3e1 \
-					rmd160  1370621e38736f14dab1ec540b9233906b5eaa31
+depends_lib         port:libiconv \
+                    port:openssl \
```

Comment:
if it can also build against other flavors like `libressl` this should become `path:lib/libssl.dylib:openssl`; alternatively you can add
```
PortGroup           openssl 1.0
``` 
at the top of the Portfile and remove the dependency here

Reply:
How would I test that it can build against `libressl`? Is there a way to force it to link to `libressl` rather than `openssl`? (I can't see anything about `libressl` in the [unshield GitHub](https://github.com/twogood/unshield))

Reply:
unsurprisingly you would replace `openssl` with `libressl` ;)

Reply:
I tried replacing `port:openssl` with `port:libressl`, but I get this after uninstalling openssl and unshield, then reinstalling with trace-mode:

```
sudo port uninstall openssl
... (skipping the list of packages which depend on openssl and will be broken)
Warning: Uninstall forced.  Proceeding despite dependencies.
--->  Deactivating openssl @3_1+universal
--->  Cleaning openssl
--->  Uninstalling openssl @3_1+universal
❯ sudo port uninstall unshield
❯ sudo port -t install unshield
--->  Computing dependencies for openssl
--->  Fetching archive for openssl
--->  Attempting to fetch openssl-3_1+universal.darwin_17.i386-x86_64.tbz2 from http://packages.macports.org/openssl
--->  Attempting to fetch openssl-3_1+universal.darwin_17.i386-x86_64.tbz2 from http://jog.id.packages.macports.org/macports/packages/openssl
--->  Attempting to fetch openssl-3_1+universal.darwin_17.i386-x86_64.tbz2 from http://ywg.ca.packages.macports.org/mirror/macports/packages/openssl
--->  Fetching distfiles for openssl
--->  Verifying checksums for openssl
--->  Extracting openssl
--->  Configuring openssl
--->  Building openssl
--->  Staging openssl into destroot
--->  Installing openssl @3_1+universal
--->  Activating openssl @3_1+universal
--->  Cleaning openssl
--->  Computing dependencies for unshield
Error: Can't install libressl because conflicting ports are active: openssl
Error: Follow https://guide.macports.org/#project.tickets if you believe there is a bug.
Error: Processing of port unshield failed
```

i.e. It's still trying to install `openssl`, even though it's not listed in the Portfile, then it tries to install `libressl` and has a conflict.

Reply:
OK. I manually installed `libressl` first, then ran `port -t install unshield`, and it appears to have worked!

Reply:
I'll push the updated Portfile today or tomorrow.

Reply:
@reneeotten I've tested that it builds against `libressl` successfully, and I've pushed a change to use `PortGroup openssl 1.0`. Is that right?

Reply:
you didn't change to use the openssl PG, but that's fine; just change the line `port:openssl` to `path:lib/libssl.dylib:openssl`.

That tells MacPorts to look for the `libssl.dylib` file and if it's not present install `openssl` (this file would be present if `libressl` is already installed, so it will not try to install `openssl` which would cause the "conflict" error you saw).

Reply:
I forgot to `git add Portfile` before I committed...

What is the difference between `PortGroup openssl 1.0` and using `path:lib/libssl.dylib:openssl`?
Do these both allow `libressl` support?  Or is `PortGroup openssl` just an alternative to `port:openssl`?

Reply:
there is no real difference, the PortGroup does allow for an easier way to use for example openssl v1 (the current default is openssl3). So if you want to use the PG instead (eventually we would probably want to transition ports to do so) that is perfectly fine, it will use the same way of specifying the dependency so `libressl` will also be able to satisfy the dependency.

Reply:
please take care of the remaining review comments, push these changes and we can merge this